src/main/java/org/jboss/remoting3/remote/RemoteReadListener.java

```diff
@@ -419,6 +419,7 @@ public void handleEvent(final ConduitStreamSourceChannel channel) {
                         case Protocol.SERVICE_ERROR: {
                             log.trace("Received service error");
                             int channelId = buffer.getInt() ^ 0x80000000;
+                            handler.handleOutboundChannelClosed(); // TODO how do I know it is outbound and not inbound?
```

Comment:
@dmlloyd is this correct? I looked at the code, and it looked like I could make this assumption... or there is no way of knowing if it is an outbound or inbound channel that generated the message?

Reply:
Yeah IIRC this can only happen on outbound channel open requests.